### ISSUE TITLE
Implement HeaterCooler for Fridge/Freezer

### ISSUE THREAD

**Issue by luigi-lauro** [thoukydides/homebridge-homeconnect#58]
#### Description of Enhancement
Since it's supported both by HomeKit and home connect APIs (ref: 
https://api-docs.home-connect.com/settings?#setpoint-temperature-freezer), my suggestion is to add a HeaterCooler accessory to control fridge and freezer. 

#### HomeKit Mapping
HeaterCooler for Fridge and Freezer. Temperature slider to set temperature. Mode dial to change statuses which are mutually exclusive vs normal mode (e.g. super, fresh, eco, etc...), integrated switch for additional options which are separate (e.g power, sabbath). There could be better ideas but this would definitely work and makes better sense than current implementation which is limited to statuses. 

#### Home Connect Appliance(s)
I could test it as a beta tester since I own a Siemens Fridge KG56NHX3P/05

**Comment by thoukydides** [comment]
The `Heater Cooler` service is intended to be used to control the room temperature, e.g. for a room thermostat or air conditioning unit. I had considered using it for the Oven temperature, but doing so breaks normal Siri interactions. For example, asking for the temperature in your house or room would include that when working out the min/max range, which is clearly not useful. Similarly, asking to set a room to a particular temperature would also try to set the appliance to the same temperature.

It is a similar issue with your request for a "mode dial". The only one that I am aware of that is currently supported by HomeKit is the source selector for televisions (and even that is not documented in Apple's Non-Commercial version of the HAP Specification). Using that for something that does not resemble a television ends up with Siri not behaving sensibly. It is also not obvious that each of the options supported by Fridge/Freezer/FridgeFreezer appliances are always mutually exclusive - there is nothing in the Home Connect developer documentation stating that it is not possible to enable multiple modes simultaneously.

I have stopped using other developers' plugins that have misused HomeKit services in this way, so definitely will not be implementing them in mine. Sorry.

The other features that you request are already supported. The plugin creates `Switch` services that allow the Freezer super mode, Fridge super mode, Eco Mode, Vacation mode, and Fresh mode to be controlled. The functionality is not just limited to statuses.

**Comment by luigi-lauro** [comment]
Understood, I think you may have a bit of a purist approach which makes sense not to mess voice control or similar, but limits HomeKit functionalities for other purposes. Which is fine, depends on how you use HomeKit and what you value most, for example to me the most useful part is control through home app, and considering your approach is the one that is the most respectful of Apple original intent I cannot really argue. Let's hope with matter supports and such iOS will widen the type of devices supported and give native options also for appliances which could be leverage on also by plugins. Thank you for now.

**Comment by thoukydides** [comment]
I primarily use HomeKit via voice control on HomePods and my Apple Watch. I very rarely use the Home app (or third-party alternatives), other than when configuring automations. Hence, yes, maintaining the correct semantics for voice control is my primary concern.

This plugin does add some characteristics that are not normally part of the services to which they have been added. However, I have only done that when I have confirmed that it does not break Siri functionality, and when the features being controlled or indicated in that way cannot be sensibly mapped to services that would allow voice control.

If Apple add services and characteristics that are a better match than those used currently, or that would allow additional features to be supported, then I would expect to use them.
